x-pack/plugins/security_solution/server/lib/detection_engine/signals/source_fields_merging/strategies/merge_missing_fields_with_source.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get } from 'lodash/fp';
+import { set } from '@elastic/safer-lodash-set/fp';
+import { SignalSource, SignalSourceHit } from '../../types';
+import { filterFieldEntries } from '../utils/filter_field_entries';
+import type { FieldsType } from '../types';
+import { recursiveUnboxingFields } from '../utils/recursive_unboxing_fields';
+import { isTypeObject } from '../utils/is_type_object';
+import { arrayInPathExists } from '../utils/array_in_path_exists';
+import { isNestedObject } from '../utils/is_nested_object';
+
+/**
+ * Merges only missing sections of "doc._source" with its "doc.fields" on a "best effort" basis. See ../README.md for more information
+ * on this function and the general strategies.
+ * @param doc The document with "_source" and "fields"
+ * @param throwOnFailSafe Defaults to false, but if set to true it will cause a throw if the fail safe is triggered to indicate we need to add a new explicit test condition
+ * @returns The two merged together in one object where we can
+ */
+export const mergeMissingFieldsWithSource = ({
```

Comment:
Very clear to follow, and much more concise than the `mergeAllFieldsWithSource`. Nice abstractions here, makes this very complex piece a bit easier to grok 🙂